### PR TITLE
Log every API request

### PR DESCRIPTION
**Pull request by MartinEden**
https://vimc.myjetbrains.com/youtrack/issue/VIMC-463

### CODE REVIEW

File: src/blackboxTests/src/test/kotlin/org/vaccineimpact/api/blackboxTests/IndexTests.kt [hunk 1]
```diff
@@ -3,9 +3,10 @@ package org.vaccineimpact.api.blackboxTests
 import org.junit.Test
 import org.vaccineimpact.api.blackboxTests.schemas.JSONSchema
 import org.vaccineimpact.api.blackboxTests.helpers.RequestHelper
+import org.vaccineimpact.api.test_helpers.DatabaseTest
 import org.vaccineimpact.api.test_helpers.MontaguTests
 
-class IndexTests : MontaguTests()
+class IndexTests : DatabaseTest()
 {
```
Comment (MartinEden) [inline src/blackboxTests/src/test/kotlin/org/vaccineimpact/api/blackboxTests/IndexTests.kt]: Now that the API is using the database on every request, this blackbox test must set up the database properly.

File: src/config/db_version [hunk 2]
```diff
@@ -1 +1 @@
-7f67e66
\ No newline at end of file
+i463
```
Comment (MartinEden) [inline src/config/db_version]: Once you have merge in the database PR this should be updated to point at the new hash of db master.

File: src/blackboxTests/src/test/kotlin/org/vaccineimpact/api/blackboxTests/AccessLogTests.kt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+        assertThat(entry.what).`as`("What").isEqualTo("/v1" + what)
+        assertThat(entry.result).`as`("Result").isEqualTo(result)
+        assertThat(IPAddress.isValid(entry.ipAddress)).`as`("IP Address is valid").isTrue()
+    }
+
+    data class Entry
+    @ConstructorProperties("who", "timestamp", "what", "result", "ipAddress")
+    constructor(val who: String?, val timestamp: Timestamp, val what: String, val result: Int?, val ipAddress: String?)
+}
```
Comment (littledice) [inline src/blackboxTests/src/test/kotlin/org/vaccineimpact/api/blackboxTests/AccessLogTests.kt]: line 81 expects this data class to have a val called `when` - the code is failing to compile at the moment

Comment (MartinEden) [inline src/blackboxTests/src/test/kotlin/org/vaccineimpact/api/blackboxTests/AccessLogTests.kt]: Huh. Not sure how I missed that change. IDEA's compiler letting me down! Fixed now.

File: src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/AuthenticationController.kt [hunk 4]
```diff
@@ -26,8 +27,10 @@ class AuthenticationController(context: ControllerContext, htmlFormHelpers: Form
             oneRepoEndpoint("authenticate/", this::authenticate, repos.user, HttpMethod.post)
                     .withAdditionalSetup({ url, _ -> setupSecurity(url) })
```
Comment (littledice) [inline src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/AuthenticationController.kt]: I'm now seeing a compiler error on this line! passed function doesn't match expected signature

**Comment by MartinEden** [comment]
Yes, it was a merge issue. Fixed now hopefully.

**Pull request merged**